localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -1431,6 +1431,8 @@ def prepare_should_deploy_change(self, resource_id, change, stack, new_resources
             res_change["_deployed"] = is_deployed
             if not is_deployed:
                 return True
+            if action == "Add":
```

Comment:
Adding this here to ensure we have proper logic and avoid re-creation of resources - a companion PR with a test will follow soon upstream in -ext.